site/components/src/main/java/uk/nhs/hee/web/components/MiniHubComponent.java

```diff
@@ -1,61 +1,69 @@
-            boolean accessWithGuidancePath = request.getRequestContext().getResolvedSiteMapItem().getHstSiteMapItem().isWildCard();
-            Guidance previousGuidance = null, nextGuidance = null, currentGuidance = null;
-            boolean accessFromRootHub = false;
-            List<Guidance> guidancePages = miniHub.getGuidancePages();
-
-            if (accessWithGuidancePath) {
-                // The guidance name in URL will be resolved as "1" parameter name
-                String guidanceName = (String) request.getRequestContext().getResolvedSiteMapItem().getLocalParameters().get("1");
-                for (int i = 0; i < guidancePages.size(); i++) {
-                    Guidance guidance = guidancePages.get(i);
-                    if (guidance.getName().equalsIgnoreCase(guidanceName)) {
-                        currentGuidance = guidance;
-                        if (i > 0) {
-                            previousGuidance = guidancePages.get(i - 1);
-                        }
-                        if (i < guidancePages.size() - 1) {
-                            nextGuidance = guidancePages.get(i + 1);
-                        }
-                        break;
-                    }
-                }
-            } else {
-                accessFromRootHub = true;
-                // There is restriction in the CMS to make sure having at least one guidance on hub
-                currentGuidance = guidancePages.get(0);
-                if (guidancePages.size() > 1) {
-                    nextGuidance = guidancePages.get(1);
-                }
+        List<MinihubSection> miniHubSections = getMiniHubSections(document);
+        request.setModel("miniHubSections", miniHubSections);
 
-                String minihubName = request.getRequestContext().getResolvedSiteMapItem().getHstSiteMapItem().getValue();
-                request.setModel("minihubName", minihubName);
-            }
+        MinihubSection currentSection = getCurrentSection(request, miniHubSections);
+        request.setModel("currentSection", currentSection);
+    }
 
-            request.setModel("previousGuidance", previousGuidance);
-            request.setModel("currentGuidance", currentGuidance);
-            request.setModel("nextGuidance", nextGuidance);
-            request.setModel("accessFromRootHub", accessFromRootHub);
+    /**
+     * Gets siblings of current document of type MinihubSection
+     *
+     * @param document the {@link HippoBean} instance.
+     * @return the {@link List <MinihubSection>} instance sorted by Display Name
+     */
+    List<MinihubSection> getMiniHubSections(HippoBean document) {
+        List<MinihubSection> miniHubSections = document.getParentBean().getChildBeans(MinihubSection.class);
+        miniHubSections.sort((section1, section2) -> section1.getDisplayName().compareTo(section2.getDisplayName()));
```

Comment:
Perhaps it would be more readable if we use Comparator.comparing and method reference like this:
```
Comparator.comparing(HippoItem::getDisplayName)
```

Reply:
Had difficulty using Comparator cleanly - do you have a fuller example?
Using the lambda saves the extra import?

Reply:
This is how we can integrate the comparator in the code.

```
    /**
     * Gets siblings of current document of type MinihubSection
     *
     * @param document the {@link HippoBean} instance.
     * @return the {@link List <MinihubSection>} instance sorted by Display Name
     */
    List<MinihubSection> getMiniHubSections(HippoBean document) {
        List<MinihubSection> miniHubSections = document.getParentBean().getChildBeans(MinihubSection.class);
        miniHubSections.sort(Comparator.comparing(HippoItem::getDisplayName));
        return miniHubSections;
    }
```

Reply:
Thanks, done.